support/cdi/src/test/java/org/apache/shiro/cdi/AnnotatedTypeWrapperTest.java

```diff
@@ -101,9 +103,23 @@ void twoAdditionalAnnotations() {
     @Test
     void removeAnnotations() {
         initializeStubs();
-        var wrapper = new AnnotatedTypeWrapper<>(annotatedType, true,
-                Set.of(SessionScopedAnnotated.class.getDeclaredAnnotations()[0]),
-                Set.of(Annotated.class.getDeclaredAnnotations()[1]));
+        Set<Annotation> s1 = null;
+        Set<Annotation> s2 = null;
+        for (Annotation annotation: SessionScopedAnnotated.class.getDeclaredAnnotations()) {
+            if (annotation.annotationType().equals(SessionScoped.class)) {
+                s1 = Set.of(annotation);
+                break;
+            }
+        }
+        for (Annotation annotation: Annotated.class.getDeclaredAnnotations()) {
+            if (annotation.annotationType().equals(RequiresGuest.class)) {
+                s2 = Set.of(annotation);
+                break;
+            }
+        }
+        assertNotNull(s1);
+        assertNotNull(s2);
```

Comment:
I'm +1 on this, but I'll add a nit/recommendation too

I'd suggest looking up the annotation using a stream instead of the for/if/break combo.

Maybe something like this (not tested at all)

```java
Arrays.stream(SessionScopedAnnotated.class.getDeclaredAnnotations())
    .filter(a -> annotation.annotationType().equals(SessionScoped.class))
    .collect(Collectors.toSet());
```

(this could be cleaned up a bit with a `findFirst` and other tweaks)


In either case (feel free to keep the loop), you could break this logic out into a utility method so that the test logic reads well.

```java
Set<Annotation> sessionScopeAnnoations = getAnnotation(Class<?> annotatedClass, Class<?> annotation);
```

Reply:
Agreed. This is a lot more code than necessary/

Reply:
Thanks for the recommendations! Fixed.